sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -1232,6 +1242,8 @@ private Mono<RxDocumentServiceResponse> query(RxDocumentServiceRequest request)
         // then it should have been populated into request context already
         // otherwise set them here with the client level policy
 
+        request.requestContext.setSessionRetryOptions(this.sessionRetryOptions);
```

Comment:
why the retryOption need to be specific set for query? I thought it should be automatically populated when construct the request? 

and why the retryOption need to be populated in the requestContext? is it based on the thinking in the future different operation may have different config?

Reply:
Fixed in next iteration - I refactored it to have the `CosmosSessionRetryOptions` instance be directly injected into `SessionTokenMismatchRetryPolicy`.